newIDE/app/src/Profile/ChangeEmailDialog.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import React, { Component } from 'react';
+import FlatButton from '../UI/FlatButton';
+import RaisedButton from '../UI/RaisedButton';
+import Dialog from '../UI/Dialog';
+import { User as FirebaseUser } from 'firebase/auth';
+import {
+  type ChangeEmailForm,
+  type AuthError,
+} from '../Utils/GDevelopServices/Authentication';
+import LeftLoader from '../UI/LeftLoader';
+import { ColumnStackLayout } from '../UI/Layout';
+import TextField from '../UI/TextField';
+
+type Props = {|
+  firebaseUser: FirebaseUser,
+  onClose: () => void,
+  onChangeEmail: (form: ChangeEmailForm) => void,
+  changeEmailInProgress: boolean,
+  error: ?AuthError,
+|};
+
+type State = {|
+  form: ChangeEmailForm,
+|};
+
+export const getEmailErrorText = (error: ?AuthError) => {
```

Comment:
should I create a `EmailField` similar to the `UsernameField` so that the validation is in one place?

Reply:
We could at least move `getEmailErrorText` to a single function in EmailField - even if we don't have a component at least we're having a single handling of the errors (and sure, some errors don't make sense for the account creation or login, but we don't care - better handle too much than too little).